### PR TITLE
polkit_gnome: move to pkgs/by-name, use finalAttrs

### PR DESCRIPTION
**Pull request by JohnRTitor**
## Description of changes

- moves from `rec`ursion to `finalAttrs`
- Move the package to `pkgs/by-name`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by iynaix — APPROVED** [review]

File: pkgs/by-name/po/polkit_gnome/package.nix [hunk 1]
```diff
@@ -1,20 +1,22 @@
 { lib, stdenv, fetchurl, polkit, gtk3, pkg-config, intltool }:
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "polkit-gnome";
   version = "0.105";
 
   src = fetchurl {
-    url = "mirror://gnome/sources/polkit-gnome/${version}/${pname}-${version}.tar.xz";
-    sha256 = "0sckmcbxyj6sbrnfc5p5lnw27ccghsid6v6wxq09mgxqcd4lk10p";
+    url = "mirror://gnome/sources/polkit-gnome/${finalAttrs.version}/${finalAttrs.pname}-${finalAttrs.version}.tar.xz";
```
Comment (eclairevoyant) [inline pkgs/by-name/po/polkit_gnome/package.nix:7]:
```suggestion
    url = "mirror://gnome/sources/polkit-gnome/${finalAttrs.version}/polkit-gnome-${finalAttrs.version}.tar.xz";
```

`finalAttrs.pname` is almost always incorrect to use.

**Review by eclairevoyant — DISMISSED** [review]

**Comment by JohnRTitor** [comment]
Well this is definitely not supposed to be executed directly from the terminal, but rather it should be executed as a systemd service.

My original motivation for this PR was this, since a lot of people use [the full path](https://github.com/search?q=%24%7Bpkgs.polkit_gnome%7D%2Flibexec%2Fpolkit-gnome-authentication-agent-1&type=code) to execute it as a systemd service, it would look cleaner if `lib.getExe` was used.

File: pkgs/by-name/po/polkit_gnome/package.nix (hunk 1 above)
Comment (JohnRTitor) [reply pkgs/by-name/po/polkit_gnome/package.nix]: Done. Please check.

**Comment by JohnRTitor** [comment]
@eclairevoyant and others... as I have stated my points above, only use case of this package is for it direct execution as a systemd service and to ease that process. NixOS is not FHS compliant at all, so should we even care?

**Comment by eclairevoyant** [comment]
`getExe` is simply designed for executables in `$out/bin` or `$bin/bin`:

https://github.com/NixOS/nixpkgs/blob/7acc08c9833e876447675f2a89e1a154e22fa6bc/lib/meta.nix#L185-L192

Same goes for `meta.mainProgram`:

https://github.com/NixOS/nixpkgs/blob/51cd84ba57e5acde737d71919b1728fb309b1296/pkgs/README.md?plain=1#L385-L391

These apply to executables that are meant to be run directly from a terminal, e.g. via `nix run`. I still disagree with restructuring the package just to shoehorn it into a format for using `getExe`, as this executable is not meant to be run in this way; the executable is designed to operate within a specific desktop environment, and is simply an internal implementation detail. Rolling-your-own-DE (as in the case of using a WM) isn't the main usecase for this executable, so exposing it in `$out/bin` doesn't make sense IMO.

**Review by eclairevoyant — APPROVED** [review]

**Review by jtojnar — APPROVED** [review]

**Pull request merged**